autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/SectionSelect.tsx

```diff
@@ -73,43 +157,87 @@ const SectionSelect: React.FC<SectionSelectProps> = ({ id }): JSX.Element => {
     );
   }
 
-  let numSelected = 0;
   /**
    * Makes a list of `SectionInfo` elements, one for each section of this course, by iterating over
    * each section in `sections`. As it iterates, this function groups consecutive sections with the
    * same professor and honors status together inside one `<ul>` and under one header. Having them
    * all inside the same `<ul>` is important in order to get smooth transitions with sticky headers.
    */
+  let allSelected = true;
   const makeList = (): JSX.Element[] => {
+    /**
+     * Function to filter based off of selected filters.
+     * We need to use this here and in professor group,
+     * since professor group only gets a start and end index.
+     */
+    const filterSections = (sectionData: SectionSelected): boolean => {
+      const toBool = (filter: SectionFilter | undefined, val: boolean): boolean => {
+        if (!filter || filter === SectionFilter.NO_PREFERENCE) {
+          return true;
+        }
+        return (filter === SectionFilter.ONLY) ? val : !val;
+      };
+      // filter by whatever
+      const { section } = sectionData;
+
+      return toBool(honors, section.honors)
+        && toBool(remote, section.remote)
+        && toBool(asynchronous, section.asynchronous)
+        && ((section.currentEnrollment < section.maxEnrollment) || includeFull);
+    };
+
     let lastProf: string = null;
     let lastHonors = false;
     let currProfGroupStart = 0;
-    return sections.map((sectionData, secIdx) => {
-      const firstInProfGroup = lastProf !== sectionData.section.instructor.name
+    // since we will be filtering, we need to store the index somewhere
+    return sections
+      .map((sectionData, secIdx) => ({ sectionData, secIdx }))
+      .filter(({ sectionData }) => filterSections(sectionData)).map(({ sectionData, secIdx }) => {
```

Comment:
```suggestion
    return sections
      // Remember original index of sections so that ProfessorGroup uses the correct indices
      .map((sectionData, secIdx) => ({ sectionData, secIdx }))
      .filter(({ sectionData }) => filterSections(sectionData))
      .map(({ sectionData, secIdx }) => {
```
I tried rewriting this because it looked messy before realizing there was actually a reason you kept the indices, make sure to add comments for minor stuff like this so that other people can understand what your code is doing better. Also make sure to keep your indentation consistent to help with readability.

Reply:
Sounds good thanks.